### PR TITLE
MM-17694: Migrate to circleci.

### PR DESCRIPTION
**Pull request by metanerd**
Still probably needs some adjustments especially for the aws s3 upload part. 

https://mattermost.atlassian.net/browse/MM-17694
https://github.com/mattermost/mattermost-server/issues/11853

### CODE REVIEW

**Comment by cpanato** [comment]
I've created the `matterbuild-docs` context, all setup

**Comment by cpanato** [comment]
@metanerd can you please amend the commit to trigger the job? I've enabled that on circleci now

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  deploy:
+    docker:
+      - image: circleci/python:3.6
+    steps:
+      - attach_workspace:
+          at: .
+      - aws-s3/sync:
+          from: build/html
+          to: 's3://docs.mattermost.com/'
+          aws-region: AWS_DEFAULT_REGION
```
Comment (cpanato) [inline .circleci/config.yml]: you dont need to set this

Comment (metanerd) [reply .circleci/config.yml]: Updated, please have another look.

**Pull request merged**